src/managed_process.rs

```diff
@@ -26,7 +26,53 @@ fn seconds_since_epoch() -> f64 {
 pub struct ManagedProcess {
     kill_signal: Arc<Mutex<u32>>,
     pid:         Arc<Mutex<Option<u32>>>,
-    thread:      JoinHandle<()>
+    thread:      JoinHandle<()>,
+    backoff:     Arc<RwLock<Backoff>>,
+}
+
+struct Backoff {
+    restart_count: u64,
+    restart_threshold: f64,
+    start_time: f64,
+    end_time: f64,
```

Comment:
Can you use Duration/SystemTime instead of f64 for the various times? That would make the code clearer.